libs/corda-sdk/src/main/kotlin/net/corda/sdk/profile/ProfileUtils.kt

```diff
@@ -82,6 +83,28 @@ object ProfileUtils {
         return requireNotNull(profile) { "Profile with name $profileName does not exist." }
     }
 
+    fun createPropertiesMapFromCliArguments(properties: Array<String>): MutableMap<String, String> {
```

Comment:
I wonder if it would be cleaner to have a function signature like:
```suggestion
    fun createPropertiesMapFromCliArguments(properties: Set<String>): Map<String, String> {
```
so that it will have more immutable constructs, bringing it closer to the pure function.

Reply:
Ah, nice point, made this change. Thanks!